src/room_garden.c

```diff
@@ -24,20 +24,40 @@
 #include "player_data.h"
 #include "dungeon_data.h"
 #include "thing_data.h"
+#include "game_legacy.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 /******************************************************************************/
-DLLIMPORT long _DK_remove_food_from_food_room_if_possible(struct Thing *thing);
 
 /******************************************************************************/
 #ifdef __cplusplus
 }
 #endif
 /******************************************************************************/
-long remove_food_from_food_room_if_possible(struct Thing *thing)
+TbBool remove_food_from_food_room_if_possible(struct Thing *thing)
 {
-  return _DK_remove_food_from_food_room_if_possible(thing);
+    // return _DK_remove_food_from_food_room_if_possible(thing);
+    struct Room *room;
+    if ( thing->owner == game.neutral_player_num )
+    {
+        return false;
+    }
+    if ( *(short *)&thing->byte_13 != -1 )
```

Comment:
should be replaced with `thing->food.word_13`
and that `word_13` thing should be renamed to something resonable

Reply:
Are you sure? The decompiled code says byte_13, which is discrete from word_13.

Reply:
union is exactly "many thing at same location"

Reply:
Yes, but there's a byte->13 and a word_13 at different locations in a thing structure.

Reply:
Oh, hang on; there's a food struct in a thing struct?

Reply:
I checked my disassembled code, and it seems that you're right.